src/sentry/eventstore/snuba/backend.py

```diff
@@ -1,10 +1,59 @@
 from __future__ import absolute_import
 
+from datetime import datetime
 
+from sentry.utils import snuba
+from sentry.models import SnubaEvent
 from sentry.eventstore.base import EventStorage
 
+DEFAULT_START = datetime.utcfromtimestamp(0)  # will be clamped to project retention
+DEFAULT_END = datetime.utcnow()  # will be clamped to project retention
+DEFAULT_ORDERBY = ['-timestamp', '-event_id']
+DEFAULT_LIMIT = 100
+DEFAULT_OFFSET = 0
+
 
 class SnubaEventStorage(EventStorage):
     """
     Eventstore backend backed by Snuba
     """
+
+    def get_events(
+        self,
+        start=DEFAULT_START,
+        end=DEFAULT_END,
+        additional_columns=None,
+        conditions=None,
```

Comment:
I am skeptic about exposing the snuba conditions as they are through event store, which basically means Clickhouse functions. Do you already have an idea of how would this be used in production today? Asking so we can figure out whether we can provide a higher level abstraction focused on the actual use cases we have.

Reply:
Yeah. Nearly all of the endpoints that yield collections of events are using this function currently to prepare the conditions for snuba. https://github.com/getsentry/sentry/blob/master/src/sentry/api/event_search.py#L610-L664 outputs the conditions, filter_keys which are inputs to this. I'm not sure what future of this should be, but I tried to align the arguments of `get_events` with the output `event_search.get_snuba_query_args` that since it's peppered all over our codebase today and I didn't want to address changing it.

Reply:
It would be nice if the higher level abstractions didn't need to know about the difference between conditions and filter_keys.

Reply:
> It would be nice if the higher level abstractions didn't need to know about the difference between conditions and filter_keys.

+100

Reply:
Ok, I will address this in a follow up